src/persist-client/src/impl/state.rs

```diff
@@ -104,29 +133,32 @@ where
         Continue(())
     }
 
+    /// Downgrade the shard since if the presented `since_handle_id` is the
+    /// currently established `active_since_handle` of this shard. This will
+    /// fail and return the current `active_since_handle` if the presented
+    /// `since_handle_id` has been fenced off.
+    // WIP: Am I using Infallible and the Result correctly here? I think not.
```

Comment:
hmmm this is actually subtle. the Break variant of ControlFlow is used for things that are not idempotent. if e.g. this only accepted a new fence token if it was _greater_ than anything it had previously seen, then this would be idempotent. as written, i'll have to think more about it. I'm curious if the adaptor fence tokens that @benesch mentioned have an ordering